spec/components/npq_separation/admin/statement_selector_component_spec.rb

```diff
@@ -5,19 +5,34 @@
 RSpec.describe NpqSeparation::Admin::StatementSelectorComponent, type: :component do
   include StatementHelper
 
-  let(:lead_provider)    { create :lead_provider }
-  let!(:statement)       { create(:statement, lead_provider:) }
-  let(:cohort)           { statement.cohort }
-  let(:instance)         { described_class.new(statement) }
-
+  let(:lead_provider) { create :lead_provider }
+  let!(:statement) { create(:statement, lead_provider:) }
+  let!(:statement_other_cohort) { create(:statement, lead_provider:, cohort: other_cohort) }
+  let!(:statement_other_lead_provider) { create(:statement, lead_provider: create(:lead_provider)) }
```

Comment:
Minor nit but I'd prefer to see the `let`s called in the specs that need them (or in a `before`) rather than using `let!` - our linter doesn't pick this up but probably should do. Default rubocop-rspec config does reject them.

`let!` can lead to non-obvious execution flows of specs and was the origin of a lot of yesterdays CI failures (that and use of `travel_to` outside of before's)